src/scripts/Build-Bicep.ps1

```diff
@@ -1,200 +1,227 @@
+
+    .EXAMPLE
+        ./Build-Bicep ../bicep-registry/module-name
+        Generates separate modules for each supported scope for the specified module.
+
+    .EXAMPLE
+        ./Build-Bicep ../bicep-registry/module-name -Scope subscription
+        Generates the module for only one scope (subscription in this case).
+
+    .EXAMPLE
+        ./Build-Bicep ../bicep-registry/module-name -Scope subscription -CopySupportingFiles
+        Generates module for the subscription scope and copies associated files.
 #>
-Param (
-    [Parameter(Position = 0)][string] $Module,
-    [string] $Scope
+[CmdletBinding()]
+param
+(
+    [Parameter(Mandatory = $true)]
+    [ValidateScript({Test-Path -Path $_})]
+    [string]
+    $Module,
+
+    [Parameter()]
+    [ValidateSet('Subscription', 'ResourceGroup', 'ManagementGroup', 'Tenant')]
+    [string[]]
+    $Scopes = ('Subscription', 'ResourceGroup', 'ManagementGroup', 'Tenant'),
+
+    [Parameter()]
+    [switch]
+    $CopySupportingFiles,
+
+    [Parameter()]
+    [switch]
+    $IncludeTests,
+
+    [Parameter()]
+    [string]
+    $OutputPath = "../Release"
 )
 
-# Use the debug flag from common parameters to determine whether to run in debug mode
-$Debug = $DebugPreference -eq "Continue"
+function Get-NextEndRegionIndex
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [psobject[]]
+        $Content,
+
+        [Parameter(Mandatory = $true)]
+        [int]
+        $StartingIndex
+    )
 
-$outdir = "../../release"
-$scopeList = '(subscription|resourceGroup|managementGroup|tenant)';
-$scopeDirective = "//(\s*@$scopeList)+";
-$dir = Get-Item $Module;
+    for ($i = $StartingIndex; $i -lt $Content.Count; $i++)
+    {
+        if ($Content[$i] -match '\*\/')
+        {
+            return $i
+        }
+    }
+}
 
-# List of supported scopes to be updated later
-$script:scopes = @()
+$moduleName = [System.IO.Path]::GetFileNameWithoutExtension($Module)
+$parentFolderPath = Split-Path -Path $Module -Parent
+$parentFolderName = Split-Path -Path $parentFolderPath -Leaf
+$lines = Get-Content -Path $module
+$closingComment = "\*\/"
 
-# Generates modules for each supported scope
-function Build-Modules([string] $Path, [switch] $CopySupportingFiles) {
-    # Confirm path
-    if (-not (Test-Path (Join-Path $Module $Path))) {
-        return;
+# Matches any scope
+$allScopes = 'Subscription', 'ResourceGroup', 'ManagementGroup', 'Tenant'
+
+foreach ($scope in $scopes)
+{
+    $genericScopeString = "\/\*\s@($($allScopes.Where({$_ -ne $scope}) -join '|'))+"
+
+    # Setup output directory
+    $outputDirectory = Join-Path -Path $OutputPath -ChildPath "$Scope-$parentFolderName"
+    if (-not (Test-Path -Path $outputDirectory))
+    {
+        $outputDirectory = (New-Item -Path $outputDirectory -ItemType 'Directory').FullName
     }
+    $outputFileName = Join-Path -Path $outputDirectory -ChildPath "$moduleName.bicep"
```

Comment:
The files must be `main.bicep` and `test/main.test.bicep`

Reply:
This script will be used for all modules we want to build, so it will get called in the build pipeline with some basic logic that will build all of the modules without modifying the script

Reply:
This script is only necessary for Bicep Registry modules. I don't foresee needing this for anything else.